test/buffers_utest.cpp

```diff
@@ -362,3 +362,176 @@ TYPED_TEST(BuffersTest, TestCalculateSize) // NOLINT
         ASSERT_EQ(t.expected[id], new_size);
     }
 }
+
+TYPED_TEST(BuffersTest, TestGetValueAndMeta) // NOLINT
```

Comment:
If possible, try to have two test functions: one to test values, and one to test metadata (it will make test more like unit test, since you'll be testing a single function and also the test will be shorter => more readable).